src/cpuid/src/intel/normalize.rs

```diff
@@ -296,11 +296,211 @@ impl super::IntelCpuid {
 
     /// Update brand string entry
     fn update_brand_string_entry(&mut self) -> Result<(), NormalizeCpuidError> {
+        // Get host brand string.
+        let host_brand_string: [u8; BRAND_STRING_LENGTH] = crate::host_brand_string();
+
         let default_brand_string =
-            Self::default_brand_string().map_err(NormalizeCpuidError::GetBrandString)?;
+            default_brand_string(host_brand_string).map_err(NormalizeCpuidError::GetBrandString)?;
 
         self.apply_brand_string(&default_brand_string)
             .map_err(NormalizeCpuidError::ApplyBrandString)?;
         Ok(())
     }
 }
+
+/// Error type for [`IntelCpuid::default_brand_string`].
+#[derive(Debug, Eq, PartialEq, thiserror::Error)]
+pub enum DefaultBrandStringError {
+    /// Missing frequency.
+    #[error("Missing frequency: {0:?}.")]
+    MissingFrequency([u8; crate::BRAND_STRING_LENGTH]),
+    /// Missing space.
+    #[error("Missing space: {0:?}.")]
+    MissingSpace([u8; crate::BRAND_STRING_LENGTH]),
+    /// Insufficient space in brand string.
+    #[error("Insufficient space in brand string.")]
+    Overflow,
+}
+
+/// Gets the brand string always used for Intel.
+///
+/// # Errors
+///
+/// When unable to parse the host brand string.
+/// `brand_string.try_into().unwrap()` cannot panic as we know
+/// `brand_string.len() == BRAND_STRING_LENGTH`
+///
+/// # Panics
+///
+/// Never.
+// As we pass through host frequency, we require CPUID and thus `cfg(cpuid)`.
+// TODO: Use `split_array_ref`
+// (https://github.com/firecracker-microvm/firecracker/issues/3347)
+#[allow(
+    clippy::indexing_slicing,
+    clippy::integer_arithmetic,
+    clippy::arithmetic_side_effects
+)]
+#[inline]
+fn default_brand_string(
+    // Host brand string.
+    // This should look like b"Intel(R) Xeon(R) Platinum 8275CL CPU @ 3.00GHz".
+    host_brand_string: [u8; BRAND_STRING_LENGTH],
+) -> Result<[u8; BRAND_STRING_LENGTH], DefaultBrandStringError> {
+    /// We always use this brand string.
+    const DEFAULT_BRAND_STRING_BASE: &[u8] = b"Intel(R) Xeon(R) Processor @";
+
+    // The slice of the host string before the frequency suffix
+    // e.g. b"Intel(4) Xeon(R) Processor @ 3.00" and "GHz"
```

Comment:
is the `(4)` a typo? should it be `(R)`?

Reply:
Yes, fixed.

Reply:
Yes, fixed.

Reply:
I still see it here and in line 370. Not a blocker as the typo was there already.

Reply:
Nvm I see the fix is in a later commit.